### ISSUE TITLE
Feature Request: More convenient note searching

### ISSUE THREAD

**Issue by blurm** [champloohq/tusk#22]
Users can fuzzy search all the notes (or based on current notebook) based on title.
The suggestions to be selected will be updated asynchronously.

Like this:
![image](https://user-images.githubusercontent.com/10203631/30778761-977cccf2-a110-11e7-85a7-d797d1de2a30.png)

I don't know if this is possible for Tusk's current structure. It looks like most of Tusk's features are implemented by manipulating html element on web evernote. For fetch all notes' metadata might need accessing evernote SDK api. It would be a lot of workload.

**Comment by bozzmob** [comment]
One solution to this can be, in case of offline notebooks, since it can all be cached on disk. We can still perform fuzzy search operation I believe. Consider using [Fuse](https://github.com/krisk/Fuse).

**Comment by blurm** [comment]
@bozzmob I think offline notebooks would be another story. In order to implement that, Tusk need to display and edit note locally, can't use web evernote anymore. It would be a great feature and it takes times

But fuzzy searching could be easier. Fetch all the metadatas, create note's URL and link back to web evernote, let it deal with the editing.

**Comment by klauscfhq** [comment]
Yep, it would not be that hard to implement, using something along the lines of second method that @blurm described

**Comment by bozzmob** [comment]
@blurm @klauscfhq Yeah. I get it.

**Comment by klauscfhq** [comment]
The Gitter channel will be on really soon and you are super welcome to come over anytime and share your thoughts on features, bugs, improvements and the app in general : )

**Comment by klauscfhq** [comment]
New [`Gitter`](https://gitter.im/klauscfhq/tusk) channel is on : )

**Comment by andygikling** [comment]
Bump, I agree.  Tusk needs this.

It would be great to be able to search within a single note.  Seems like if the note is already loaded from the web, we should be able to have a note-only search on the bottom right or top right corner of the note (or something).  Ctrl+Shift+F perhaps?

When you search a string in the global search, you find the notes that contain your string, but it's sorta useless because the you have to then scroll through the note, looking for yellow highlights (on Sepia theme this is a problem).  It doesn't even bring you to the first occurrence of the string in the note you select...  It's a bit frustrating - it makes doing a search very laborious when you have lots of long notes.  

A note-only search would work around the issue until you figure out the long-term plan for offline note search etc.

**Comment by timoc** [comment]
Can I suggest two different helpers - a search fastkey for pre-defined searches (e.g. Alt-F brings up a list of saved searches), and a query-builder option that uses the advanced search syntax. Once this is in place, you have the ability to add something like an offline:<> tag or similar to your query builder for local file searches
